src/cmdc_tools/datasets/official/NV/data.py

```diff
@@ -0,0 +1,287 @@
+            renamed.dt = pd.to_datetime(renamed.dt)
+            return renamed.melt(id_vars=["dt"], var_name="variable_name").assign(
+                vintage=pd.Timestamp.utcnow(), fips=self.state_fips
+            )
+
+    async def _get_labels_from_graph(self, page):
+        # wait for graph to load
+        visual_modern = await page.waitForXPath("//*[@class='cartesianChart']")
+        # get all graph points
+        elems = await visual_modern.Jx(
+            "//*[@class='series']//*[@class='column setFocusRing']"
+        )
+        # print("elems\n", elems)
+        labels = [
+            (await page.evaluate("(el) => el.getAttribute('aria-label')", e))
+            for e in elems
+        ]
+
+        return labels
+
+    async def _get_cases_async(self):
+        async with with_page() as page:
+            await page.goto(self.source)
+            # Wait for dashboard to load
+            # Get next page button
+            await page.waitForXPath("//span[text()='COVID-19 ']")
+            button = await _get_next_page_button(page)
+            await button.click()
+            # Wait for dashboard to load
+            # Go to next page
+            await page.waitForXPath("//div[text()='COVID-19 Statistics by County']")
+            await button.click()
+            # Wait for dashboard to laod
+            # Go to next page
+            await page.waitForXPath("//div[text()='Results Filter for Demographics']")
+            await button.click()
+
+            await page.waitForXPath("//div[text()='Cumulative Tests Reported']")
+            await button.click()
+
+            # Find cumulative cases reported graph
+            graph = await page.waitForXPath("//*[@class='cartesianChart']")
+            await graph.click(button="right")
+            # Get table button
+            table_button = await page.waitForXPath("//h6[text()='Show as a table']")
+            await table_button.click()
+
+            labels = await self._get_labels_from_graph(page)
+
+            # parse labels
+            data = []
+            for label in labels:
+                split = label.split(". ")
+                date = split[0].split("Date")[1].strip() + "/2020"
+                tests = split[1].split(" ")
+                tests_type = tests[0].strip()
+                # Skip all new tests
+                if tests_type == "New":
+                    break
+                tests_num = int(tests[1][:-1].strip().replace(",", ""))
+                {"Date": date, f"{tests_type}": tests_num}
+                data.append({"Date": date, f"{tests_type}": tests_num})
+
+            df = pd.DataFrame(data)
+            renamed = df.rename(columns={"Date": "dt", "Cases": "cases_total"})
+            renamed.dt = pd.to_datetime(renamed.dt)
+            return renamed.melt(id_vars=["dt"], var_name="variable_name").assign(
+                vintage=pd.Timestamp.utcnow(), fips=self.state_fips
```

Comment:
Please use the `_retrieve_vintage` method so that, if we changed how we collected vintages, then it would be a change to one method rather than hunting down every place we used it.